lib/pangea/extensions/pangea_room_extension.dart

```diff
@@ -808,6 +812,85 @@ extension PangeaRoom on Room {
     );
   }
 
+  Future<void> archive() async {
+    final participants = await requestParticipants();
+    final students = participants
+        .where(
+          (e) =>
+              e.powerLevel < ClassDefaultValues.powerLevelOfAdmin &&
+              e.id != BotName.byEnvironment,
+        )
+        .toList();
+    for (final student in students) {
+      await kick(student.id);
+    }
+    if (!isSpace && membership == Membership.join && isUnread) {
+      await markUnread(false);
+    }
+    await leave();
+  }
+
+  Future<bool> archiveSpace(BuildContext context, Client client) async {
+    final confirmed = await showOkCancelAlertDialog(
+          useRootNavigator: false,
+          context: context,
+          title: L10n.of(context)!.areYouSure,
+          okLabel: L10n.of(context)!.yes,
+          cancelLabel: L10n.of(context)!.cancel,
+          message: L10n.of(context)!.archiveSpaceDescription,
+        ) ==
+        OkCancelResult.ok;
+    if (!confirmed) return false;
+    final success = await showFutureLoadingDialog(
+      context: context,
+      future: () async {
+        final List<Room> children = await getChildRooms();
```

Comment:
Can you filter out analytics room from this list? Probably do the filtering in the getChildRooms function